ml/pre-processing/get_VT_signatures.scala

```diff
@@ -0,0 +1,45 @@
+import com.datastax.spark.connector._
+import play.api.libs.json.Json
+import java.io.{ByteArrayOutputStream, ByteArrayInputStream}
+import java.util.zip.{GZIPOutputStream, GZIPInputStream}
+
+case class VT_results_by_service_name_class(service_name: String, sha256: String)
+case class VT_results_by_sha256_class(sha256: String, service_name: String, results: Array[Byte] )
+case class VT_join_results_class(sha256: String, service_name: String, results: String)
+case class VT_sample_signatures_initial_seq_rdd_class(sha256: String, seq_results: Seq[String])
+case class VT_sample_signatures_final_array_rdd_class(sha256:String, array_results:Array[Double])
+
+def unzip(x: Array[Byte]) : String = {		
+    val inputStream = new GZIPInputStream(new ByteArrayInputStream(x))
+    val output = scala.io.Source.fromInputStream(inputStream).mkString
+    return output
+}
+def deleteNumberInSampleSignatures(x: String): Boolean = {
+    val regex = "[0-9]".r
+    return regex.findFirstIn(x).isEmpty
+}
+
+val VT_results_by_service_name_meta = sc.cassandraTable[VT_results_by_service_name_class]("gsoc3","results_meta_by_service_name").where("service_name=?","virustotal")
+val VT_results_by_service_name_rdd = VT_results_by_service_name_meta.keyBy(x=> (x.sha256,x.service_name))
+val VT_results_by_sha256_meta = sc.cassandraTable[VT_results_by_sha256_class]("gsoc3","results_data_by_sha256")
+val VT_results_by_sha256_rdd = VT_results_by_sha256_meta.keyBy(x => (x.sha256,x.service_name))
+val VT_join_results = VT_results_by_service_name_rdd.join(VT_results_by_sha256_rdd).map(x => (new VT_join_results_class(x._1._1,x._1._2, unzip(x._2._2.results)))).distinct().cache()
+val sample_signatures_rdd = VT_join_results.flatMap(x=>Json.parse(x.results) \ "scans" \\ "result").map(x=>Json.stringify(x)).filter( x=> !(x == "null"))
+val sample_signatures_split_rdd = sample_signatures_rdd.flatMap(x=>x.replaceAll("""["]""","").replaceAll("""\![a-zA-Z0-9\s\+]+""","").replaceAll("""@[a-zA-Z0-9\s\+]+""","").replaceAll("""~[a-zA-Z0-9\s\+]+""","").replaceAll("""[\(|\[|{][a-zA-Z0-9\s\+]*[\)|\]|}]""","").replaceAll("""(\.|\!|\:|\_|\-|\\|/|\[|\])"""," ").split(" ")).filter(x=>(x.size>3)).filter(x=>deleteNumberInSampleSignatures(x)).map(x=>x.toLowerCase())
+val signatures_prefix_rdd = sc.textFile("./t0day/prefix.txt").map(x=>x.toLowerCase())
```

Comment:
The prefix.txt file is the prefix and suffix of the VT results that Bojan sent me.
Considering some words in VT results are too normal (for example Malware, TrojWare), we should delete it. 
So this file does not need to compile and I just save this file. 
What do you think of making a new directory in ml/pre-processing to save prefix.txt file?